data-library/library.sh

```diff
@@ -16,7 +16,46 @@ function library_execute {
     fi
 }
 
+function setup {
+    apt update
+    apt-get install -y python3-pip python3-distutils gdal-bin libgdal-dev curl git
+    apt-get -y install --reinstall build-essential
+
+    python3 -m pip install poetry
+    poetry config virtualenvs.create false --local
+
+    case $1 in 
+        dev) ;;
+        *) option="--no-dev";;
+    esac
+    gdal_version=$(gdal-config --version)
+    gdal_poetry_version=$(poetry_version gdal)
+
+    if [ $gdal_version != $gdal_poetry_version ]; then
+        echo "mismatch between installed gdal binary and gdal version in pyproject.yaml"
+        echo "update poetry gdal version to $gdal_version"
+        exit 1
+    fi
+    poetry install $option
+
+    export CPLUS_INCLUDE_PATH=/usr/include/gdal
+    export C_INCLUDE_PATH=/usr/include/gdal
+}
+
+function poetry_version {
+    echo $(poetry show $1) | sed -r 's/^.*version : ([0-9\.]+) .*$/\1/'
+}
+
+function poetry_lock {
```

Comment:
This is what should be used now to regenerate lock file - ensures that python gdal version matches what's installed.